e3sm_diags/logger.py

```diff
@@ -0,0 +1,93 @@
+"""Logger module for setting up a custom logger."""
+import logging
+import logging.handlers
+import shutil
+
+LOG_FILENAME = "e3sm_diags_run.log"
+
+
+def custom_logger(name: str, propagate: bool = True) -> logging.Logger:
+    """Sets up a custom logger.
+
+    Parameters
+    ----------
+    name : str
+        Name of the file where this function is called.
+    propagate : bool, optional
+        Whether to propagate logger messages or not, by default True.
+
+    Returns
+    -------
+    logging.Logger
+        The logger.
+
+    Examples
+    ---------
+    Detailed information, typically of interest only when diagnosing problems:
+
+    >>> logger.debug("")
+
+    Confirmation that things are working as expected:
+
+    >>> logger.info("")
+
+    An indication that something unexpected happened, or indicative of some
+    problem in the near future:
+
+    >>> logger.warning("")
+
+    The software has not been able to perform some function due to a more
+    serious problem:
+
+    >>> logger.error("")
+
+    Similar to ``logger.error()``, but also outputs stack trace:
+
+    >>> logger.exception("", exc_info=True)
+
+    A serious error, indicating that the program itself may be unable to
+    continue running:
+
+    >>> logger.critical("")
+    """
+    log_format = "%(asctime)s [%(levelname)s]: %(filename)s(%(funcName)s:%(lineno)s) >> %(message)s"
+    log_filemode = "w"
+
+    # Setup
+    logging.basicConfig(
+        format=log_format,
+        filename=LOG_FILENAME,
+        filemode=log_filemode,
+        level=logging.INFO,
+    )
+    logging.captureWarnings(True)
```

Comment:
This captures warnings from the Python `warning` package (sometimes used in other libraries)